scrape/scrape_test.go

```diff
@@ -362,14 +379,44 @@ func TestScrapePoolReloadPreserveRelabeledIntervalTimeout(t *testing.T) {
 		newLoop:     newLoop,
 		logger:      nil,
 		client:      http.DefaultClient,
-		metrics:     newTestScrapeMetrics(t),
+		metrics:     metrics,
 		symbolTable: labels.NewSymbolTable(),
 	}
 
 	err := sp.reload(reloadCfg)
 	if err != nil {
 		t.Fatalf("unable to reload configuration: %s", err)
 	}
+	// Check that the reload metric is labeled with the pool interval, not the overridden interval.
+	got, err := gatherLabels(reg, "prometheus_target_reload_length_seconds")
+	require.NoError(t, err)
+	testutil.RequireEqual(t, []labels.Labels{labels.FromStrings("interval", "3s")}, got["prometheus_target_reload_length_seconds"])
+}
+
+// Gather metrics from the provided Gatherer with specified familynames,
+// and extract labels. Return for each family name, a map of name/value pairs.
+func gatherLabels(g prometheus.Gatherer, familyNames ...string) (map[string][]labels.Labels, error) {
```

Comment:
Would it be possible to reduce the indentation, it's a little bit hard to read. I see this is only called with one familyNames and could we make the util itself check for the existence of the labels?